### PR TITLE
Per-provider line items on statements

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: go/billing/tests/test_tasks.py [hunk 1]
```diff
@@ -449,12 +524,224 @@ def test_generate_monthly_statement_sessions_none_markup(self):
...
+            message_direction=MessageCost.DIRECTION_INBOUND)
+
+        statement = tasks.generate_monthly_statement(
+            self.account.id, *this_month())
+
+        [item1, item2] = get_line_items(statement).filter(
+            description__startswith='Messages received')
+
+        self.assertEqual(item1.description, 'Messages received - Provider 1')
+        self.assertEqual(item2.description, 'Messages received - Provider 2')
```
Comment (rudigiesler) [inline go/billing/tests/test_tasks.py:590]: I'm not sure if this is relevant (I'm not sure how we want the statement to be grouped), but would it be useful to test that the providers are being  grouped correctly, ie. Make sure that the quantity is 2 for `provider1` and 1 for `provider2`.

File: go/billing/settings.py [hunk 2]
```diff
@@ -56,10 +56,10 @@
 LOW_CREDIT_NOTIFICATION_EMAIL = getattr(
     settings, 'BILLING_LOW_CREDIT_NOTIFICATION_EMAIL', 'support@vumi.org')
 
-PROVIDERS = getattr(
-    settings, 'BILLING_PROVIDERS', (
-        ('mtn', 'MTN'),
-        ('vodacom', 'Vodacom')))
+PROVIDERS = getattr(settings, 'BILLING_PROVIDERS', {
+    'mtn': 'MTN',
+    'vodacom': 'Vodacom'
```
Comment (rudigiesler) [inline go/billing/settings.py]: Trailing comma?

File: go/billing/tests/test_tasks.py (hunk 1 above)
Comment (justinvdm) [inline go/billing/tests/test_tasks.py:590]: +1, will do

**Comment by rudigiesler** [comment]
:+1:

**Pull request merged**